### ISSUE TITLE
Distributional DDPG Progress?

### ISSUE THREAD

**Issue by AlexanderKoch-Koch** [astooke/rlpyt#119]
What is the status of Distributional DDPG? In the Readme, it says "coming soon".

**Comment by astooke** [comment]
Good question, it definitely seems worth including, but my immediate to-do list is already full with other things.  Can see if I can find someone to code it up, or do you want to give it a try?  I have some compute to run mujoco benchmarks to verify.

Anyone else have it up and running?

**Comment by kargarisaac** [comment]
I'm interested to do it. For now, I'm trying to discretize my environment and use r2d1. I will try to replace dqn with ddpg, after trying r2d1.

**Comment by kargarisaac** [comment]
I want to start to add the recurrent replay for DDPG. But before that, I want to understand the recurrent replay better. I'm thinking of using recurrent replay for PPO_LSTM instead of DDPG.
- I tested PPO_LSTM and the result is good. In this implementation, it collects some trajectories and uses them to train LSTM and then clean the buffer, right? It is sequentially, first collect data then update the net. In openAI five, they use async sampling and optimization (I think it is similar to async sampler and runner in rlpyt?) to collect data and save in a buffer and an optimizer samples data to do gradient descent and update worker weights. So this is not on-policy anymore and they do not erase the buffer, right? Is it okay to extend the PPO_LSTM to do sth similar?

**Comment by astooke** [comment]
Great!  Right, that is all correct about buffers + async, etc.  It seems PPO is able to handle being slightly off-policy, and could be worth a try in the async setting.  Eventually it could be interesting to compare against an algorithm like V-trace for slight off-policy learning.
